alternator/streams.cc

```diff
@@ -220,38 +222,59 @@ future<alternator::executor::request_return_type> alternator::executor::list_str
 
 struct shard_id {
     static constexpr auto marker = 'H';
-
-    utils::UUID table;
-    db_clock::time_point time;
-    cdc::stream_id id;
-
-    shard_id(utils::UUID tab, db_clock::time_point t, cdc::stream_id i)
-        : table(tab)
-        , time(t)
-        , id(i)
-    {}
+    static constexpr auto hex_width = sizeof(uint64_t)*8/4;
+
+    db_clock::time_point start_time;
+    db_clock::time_point end_time; // zero if open
```

Comment:
What does this start_time and end_time represent logically? Is start_time the time the CDC generation that this shard belongs to was opened? Is end_time the time this CDC generation was closed?

Reply:
Yes. Though "closed" is defined as "next generation was opened".